lib/configs/imports.ts

```diff
@@ -0,0 +1,127 @@
+import type { Linter } from 'eslint'
+/*!
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import type { ConfigOptions } from '../types.d.ts'
+
+import perfectionist from 'eslint-plugin-perfectionist'
+import {
+	GLOB_FILES_JAVASCRIPT,
+	GLOB_FILES_TYPESCRIPT,
+	GLOB_FILES_VUE,
+} from '../globs.ts'
+
+/**
+ * Generate imports and exports related ESLint rules.
+ *
+ * @param options - Configuration options
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function imports(options: ConfigOptions): Linter.Config[] {
+	return [
+		{
+			name: 'nextcloud/imports/setup',
+			plugins: {
+				perfectionist,
+			},
+		},
+		{
+			name: 'nextcloud/imports/rules',
+			files: [
+				...GLOB_FILES_JAVASCRIPT,
+				...GLOB_FILES_TYPESCRIPT,
+				...GLOB_FILES_VUE,
+			],
+			rules: {
+				// Require file extensions
+				'no-restricted-imports': [
+					'error',
+					{
+						patterns: [
+							{
+								regex: '^(\\.*)/(.+/)*[^/.]+$',
+								message: 'Import is missing the file extension.',
+							},
+						],
+					},
+				],
+				// Sorting of imports
+				'sort-imports': 'off',
+				'perfectionist/sort-imports': [
+					'error',
+					{
+						type: 'natural',
+						newlinesBetween: 'never',
+						groups: [
+							// type first
+							'external-type',
+							'type',
+							{ newlinesBetween: 'always' },
```

Comment:
Maybe remove these empty lines as well?

Reply:
So imports are a single block, separated by an empty line from the remaining module code.
Type imports are separated visually via `import type` prefix anyway.

Reply:
```suggestion
							{ newlinesBetween: 'ignore' },
```

Maybe?